bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/BigtableHBaseClassicSettings.java

```diff
@@ -0,0 +1,380 @@
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.MAX_ELAPSED_BACKOFF_MILLIS_KEY;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.MAX_INFLIGHT_RPCS_KEY;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.MAX_SCAN_TIMEOUT_RETRIES;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.READ_BUFFER_SIZE;
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.READ_PARTIAL_ROW_TIMEOUT_MS;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.google.api.core.InternalApi;
+import com.google.auth.Credentials;
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.config.BulkOptions;
+import com.google.cloud.bigtable.config.CallOptionsConfig;
+import com.google.cloud.bigtable.config.CredentialOptions;
+import com.google.cloud.bigtable.config.RetryOptions;
+import com.google.cloud.bigtable.hbase.BigtableExtendedConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.cloud.bigtable.hbase.wrappers.BigtableHBaseSettings;
+import com.google.common.base.Preconditions;
+import io.grpc.Status;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.util.VersionInfo;
+
+/** For internal use only - public for technical reasons. */
+@InternalApi("For internal usage only")
+public class BigtableHBaseClassicSettings extends BigtableHBaseSettings {
+
+  private final Configuration configuration;
+  private final BigtableOptions bigtableOptions;
+
+  public BigtableHBaseClassicSettings(Configuration configuration) throws IOException {
+    super(configuration);
+    // we can't create a defensive copy because it might be an instance
+    // BigtableExtendedConfiguration
+    this.configuration = configuration;
+    BigtableOptions.Builder bigtableOptionsBuilder = BigtableOptions.builder();
+
+    bigtableOptionsBuilder.setProjectId(getProjectId()).setInstanceId(getInstanceId());
+
+    String appProfileId = configuration.get(APP_PROFILE_ID_KEY);
+    if (!isNullOrEmpty(appProfileId)) {
+      bigtableOptionsBuilder.setAppProfileId(appProfileId);
+    }
+
+    configureBulkOptions(bigtableOptionsBuilder);
+    configureChannelOptions(bigtableOptionsBuilder);
+    configureClientCallOptions(bigtableOptionsBuilder);
+    configureCredentialOptions(bigtableOptionsBuilder);
+    configureRetryOptions(bigtableOptionsBuilder);
+
+    String emulatorHost = configuration.get(BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY);
+    if (!isNullOrEmpty(emulatorHost)) {
+      bigtableOptionsBuilder.enableEmulator(emulatorHost);
+    }
+
+    String useBatchStr = configuration.get(BIGTABLE_USE_BATCH);
+    if (!isNullOrEmpty(useBatchStr)) {
+      bigtableOptionsBuilder.setUseBatch(Boolean.parseBoolean(useBatchStr));
```

Comment:
please add a check here to ensure `useBatchStr` is a valid value

Reply:
I don't think we should do that...hadoop's configuration class doesn't get valid boolean values, neither does parseBoolean.

Reply:
What is `useBatchStr` expected to evaluate to?

Reply:
`useBatchStr` is expected to be a boolean, the default value for this is `false`.